docs/Integrate GitOps with Secrets Management.md

```diff
@@ -0,0 +1,423 @@
+
+    * **Installed Namespace** is set to **openshift-cluster-csi-drivers**.
+
+    g. Click **Install**.
+
+    After the installation finishes, the Secrets Store CSI Driver Operator is listed in the **Installed Operators** section of the web console.
+
+2. Create the ClusterCSIDriver instance for the driver (`secrets-store.csi.k8s.io`):
+
+    a. Click **Administration** → **CustomResourceDefinitions** → **ClusterCSIDriver**.
+
+    b. On the **Instances** tab, click **Create ClusterCSIDriver**.
+    Use the following YAML file:
+
+    ```
+    apiVersion: operator.openshift.io/v1
+    kind: ClusterCSIDriver
+    metadata:
+        name: secrets-store.csi.k8s.io
+    spec:
+    managementState: Managed
+    ```
+
+    c. Click **Create**.
+
+### Install GitOps
+1. Install the GitOps Operator
+
+Follow the steps on [Installing OpenShift GitOps](OpenShift%20GitOps%20Usage%20Guide.md#installing-openshift-gitops).
+
+2. Bootstrap a GitOps repository
+
+Follow the steps on [Getting started with GitOps Application Manager (kam)](OpenShift%20GitOps%20Usage%20Guide.md#getting-started-with-gitops-application-manager-kam) and [Day 1 Operations](https://github.com/redhat-developer/kam/tree/master/docs/journey/day1).
+
+
+### Store resources of AWS Secrets Manager in the GitOps repository
+1. Add resources for AWS Secrets Manager provider
+
+    a. In your KAM generated GitOps repository, add `secret-provider-app.yaml` for resources of AWS Secrets Manager to `/config/argocd` directory.
+
+    *Example `secret-provider-app.yaml` file*
+    ```
+    apiVersion: argoproj.io/v1alpha1
+    kind: Application
+    metadata:
+    creationTimestamp: null
+    name: secret-provider-app
+    namespace: openshift-gitops
+    spec:
+    destination:
+        namespace: openshift-cluster-csi-drivers
+        server: https://kubernetes.default.svc
+    ignoreDifferences:
+    - group: argoproj.io
+        jsonPointers:
+        - /status
+        kind: Application
+    - group: triggers.tekton.dev
+        jsonPointers:
+        - /status
+        kind: EventListener
+    - group: triggers.tekton.dev
+        jsonPointers:
+        - /status
+        kind: TriggerTemplate
+    - group: triggers.tekton.dev
+        jsonPointers:
+        - /status
+        kind: TriggerBinding
+    - group: route.openshift.io
+        jsonPointers:
+        - /spec/host
+        kind: Route
+    project: default
+    source:
+        path: config/sscsid
+        repoURL: https://github.com/your-domain/gitops.git
+    syncPolicy:
+        automated:
+        prune: true
+        selfHeal: true
+    status:
+    health: {}
+    summary: {}
+    sync:
+        comparedTo:
+        destination: {}
+        source:
+            repoURL: ""
+        status: ""
+    ```
+
+    b. Create `/config/sscsid/` directory and add `aws-provider.yaml` file in it to deploy resources for AWS Secrets Manager.
+
+    *Example `aws-provider.yaml` file*
```

Comment:
Is there a way to create these resources through operator by submitting an operand ? 

For example, you submit a AWS Provider CR to the CSI operator and it creates all the manifests that you mentioned below.

The problem that I see here is about maintenance, Any changes to these YAML manifests should be tracked and maintained.

I am also fine with a Helm chart instead of these plain YAML manifests.

Please check if there is a way.

Reply:
According to their documentation, they don't have a way to maintain these YAML manifests via Helm chart or operand yet.
https://docs.openshift.com/container-platform/4.14/nodes/pods/nodes-pods-secrets-store.html#secrets-store-aws_nodes-pods-secrets-store